app/models/project_services/bamboo_service.rb

```diff
@@ -131,7 +131,18 @@ def execute(data)
     return unless supported_events.include?(data[:object_kind])
 
     # Bamboo requires a GET and does not take any data.
-    self.class.get("#{bamboo_url}/updateAndBuild.action?buildKey=#{build_key}",
-                   verify: false)
+    url = URI.parse("#{bamboo_url}/updateAndBuild.action?buildKey=#{build_key}")
+
+    # Bamboo may require authentification even for the build trigger
+    if username.blank? && password.blank?
+      self.class.get(url, verify: false)
+    else
+      get_url = "#{url}&os_authType=basic"
```

Comment:
Did `os_authType=basic` work for you, still? What version of Bamboo? I was in the process of testing another issue with Bamboo service and had to remove this parameter for basic auth to work in 5.9.1.